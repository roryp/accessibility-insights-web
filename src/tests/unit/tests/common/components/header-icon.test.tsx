// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import * as React from 'react';

import {
    HeaderIconComponent,
    HeaderIconDeps,
    HeaderIconProps,
} from '../../../../../common/components/header-icon';

describe('HeaderIconComponent', () => {
    let props: HeaderIconProps;

    beforeEach(() => {
        props = {
            deps: {
                storeActionMessageCreator: null,
                storesHub: null,
            } as HeaderIconDeps,
            storeState: {
                userConfigurationStoreData: {
                    enableHighContrast: null,
                },
            },
        } as HeaderIconProps;
    });

    describe('renders', () => {
        describe.each([true, false, undefined])('with invertColors = %s', invertColors => {
            it.each([true, false])('and high contrast mode = %s', enableHighContrast => {
                props.invertColors = invertColors;
                props.storeState.userConfigurationStoreData.enableHighContrast = enableHighContrast;
                const wrapper = shallow(<HeaderIconComponent {...props} />);
                expect(wrapper.getElement()).toMatchSnapshot();
            });
        });
    });
});
