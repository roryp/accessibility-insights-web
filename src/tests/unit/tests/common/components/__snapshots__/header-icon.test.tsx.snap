// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderIconComponent renders with invertColors = false and high contrast mode = false 1`] = `<BrandWhite />`;

exports[`HeaderIconComponent renders with invertColors = false and high contrast mode = true 1`] = `<BrandBlue />`;

exports[`HeaderIconComponent renders with invertColors = true and high contrast mode = false 1`] = `<BrandBlue />`;

exports[`HeaderIconComponent renders with invertColors = true and high contrast mode = true 1`] = `<BrandWhite />`;

exports[`HeaderIconComponent renders with invertColors = undefined and high contrast mode = false 1`] = `<BrandWhite />`;

exports[`HeaderIconComponent renders with invertColors = undefined and high contrast mode = true 1`] = `<BrandBlue />`;
